.github/workflows/vcpkg_docker_amd64.yml

```diff
@@ -38,28 +41,42 @@ jobs:
+            test_name="test-${BRANCH_NAME////_}-"
+          fi
+          echo "IMAGE_NAME=${test_name}vcpkg-builder-ubuntu-${{ env.IMAGE_VER }}:${{ matrix.container.version }}${{ matrix.host.arch }}" >> ${GITHUB_ENV}
+
       - name: Build and Push
         working-directory: docker
         run: |
           # Pull freshest ubuntu Docker image
           docker pull ubuntu:${{ matrix.container.version}}
 
-          docker build -f Dockerfile.ubuntu.vcpkg --target caching --no-cache --build-arg "DISTRO_VERSION=${{ matrix.container.codename }}" -t "docker.pkg.github.com/${DOCKER_TAG}" .
+          # Build for GitHub Actions registry
           # Docker image with NuGet support goes to github packages for CI use only
-          if [[ "${GITHUB_REF}" == "refs/heads/master" ]] ; then
-            docker login docker.pkg.github.com -u publisher -p "${GITHUB_PACKAGE_REGISTRY_TOKEN}"
-            for i in 1 2 3; do docker push "docker.pkg.github.com/${DOCKER_TAG}" && break || sleep 10; done
-          fi
+          DOCKER_TAG="docker.pkg.github.com/lifting-bits/cxx-common/${IMAGE_NAME}"
+          docker build -f Dockerfile.ubuntu.vcpkg --no-cache \
+              --target caching \
+              --build-arg "DISTRO_VERSION=${{ matrix.container.codename }}" \
+              -t "${DOCKER_TAG}" \
+              .
+          docker login docker.pkg.github.com -u publisher -p "${GITHUB_PACKAGE_REGISTRY_TOKEN}"
+          for i in 1 2 3; do docker push "${DOCKER_TAG}" && break || sleep 10; done
 
           # NOTE: Docker Hub only allows one slash in tag
-          docker build -f Dockerfile.ubuntu.vcpkg --target base --build-arg "DISTRO_VERSION=${{ matrix.container.codename }}" -t "trailofbits/cxx-common-vcpkg-builder-${{ matrix.container.distro }}:${{ matrix.container.version }}${{ matrix.host.arch }}" .
+          DOCKER_TAG="trailofbits/cxx-common-${IMAGE_NAME}"
+          docker build -f Dockerfile.ubuntu.vcpkg --target base --build-arg "DISTRO_VERSION=${{ matrix.container.codename }}" -t "${DOCKER_TAG}" .
           # Smaller Docker image without NuGet support goes to Docker Hub for users
           if [[ "${GITHUB_REF}" == "refs/heads/master" ]] ; then
             docker login -u "${DOCKER_HUB_USER}" -p "${DOCKER_HUB_TOKEN}"
-            for i in 1 2 3; do docker push "trailofbits/cxx-common-vcpkg-builder-${{ matrix.container.distro }}:${{ matrix.container.version }}${{ matrix.host.arch }}" && break || sleep 10; done
+            for i in 1 2 3; do docker push "${DOCKER_TAG}" && break || sleep 10; done
           fi
         env:
-          DOCKER_TAG: lifting-bits/cxx-common/vcpkg-builder-${{ matrix.container.distro }}:${{ matrix.container.version }}${{ matrix.host.arch }}
           GITHUB_PACKAGE_REGISTRY_TOKEN: ${{  secrets.GITHUB_PACKAGE_REGISTRY_TOKEN  }}
-          DOCKER_HUB_USER: ${{  secrets.DOCKER_HUB_USER  }}
-          DOCKER_HUB_TOKEN: ${{  secrets.DOCKER_HUB_TOKEN  }}
+          # DOCKER_HUB_USER: ${{  secrets.DOCKER_HUB_USER  }}
```

Comment:
we can just get rid of this i think

Reply:
Ah! Good catch. I think I actually want to re-enable this. I disabled because I was scared about it pushing to Docker Hub during testing.

Reply:
Okay, I removed Docker Hub integration. This is a breaking change, but that's fine because the image is also a different name now.